tests/test_dataclasses.py

```diff
@@ -945,3 +945,30 @@ def __new__(cls, *args, **kwargs):
         instance = cls(a=test_string)
         assert instance._special_property == 1
         assert instance.a == test_string
+
+
+def test_allow_extra():
+    class C:
+        extra = Extra.ignore
+
+    @pydantic.dataclasses.dataclass(config=C)
+    class Foo:
+        a: str
+        b: str
+
+    Foo(**{"a": "bar", "b": "foo", "c": 1})
```

Comment:
* single quotes
* `Foo(a='bar'...)` would be cleaner
* you need to add `assert not hasattr(foo, 'c')`
* you need to add another test for `Extra.allow`